metasrv/src/meta_service/raftmeta.rs

```diff
@@ -412,6 +412,47 @@ impl MetaNode {
         Ok(mn)
     }
 
+    pub async fn join_cluster(&self, conf: &RaftConfig) -> MetaResult<()> {
+        if conf.join.is_empty() {
+            return Ok(());
+        }
+
+        // Try to join a cluster only when this node is just created.
+        // Joining a node with log has risk messing up the data in this node and in the target cluster.
+        if self.is_opened() {
+            return Ok(());
+        }
+
+        let addrs = &conf.join;
+        // Join cluster use advertise host instead of listen host
+        let raft_api_advertise_host_endpoint = conf.raft_api_advertise_host_endpoint();
+        #[allow(clippy::never_loop)]
```

Comment:
Why not just use `addrs[0]` ?

Reply:
It's an unfinished code snippet... There should be a retry loop... :((